### ISSUE TITLE
Possibly research token generation, replacing with static token.

### ISSUE THREAD

**Issue by balika011** [michaelcowell/xiaomi-bootloader#2]
The token is created from the cpuid (different for every singe device) and some random number.
The token is stored on the device, rsa encrypted.
On every boot the device decrypts the token and compares to the cpuid.
If it matches then the unlocked flag will be true.
-> The token can't be replaced.

(But once you have a token for a cpuid it can be written back using EDL, the device will be unlocked.)

**Comment by xiaohuzz** [comment]
![redmi note 5a prime](https://user-images.githubusercontent.com/14057911/34335284-70944582-e988-11e7-8cbc-fe0e547bf7a2.png)
![redmi note 5a](https://user-images.githubusercontent.com/14057911/34335285-70c190fa-e988-11e7-9620-637d1e66ab9c.png)

**Comment by xiaohuzz** [comment]
This is devinfo, EDL write back，
the device will be unlocked！

**Comment by balika011** [comment]
But only works on your device. Would't unlock another one.

**Comment by xiaohuzz** [comment]
No, all note5a are applicable, I've already tested it. This is already cracked!

**Comment by vlackhonor1** [comment]
hello Dear Bro, How to edit please guide me....
thanks  :)

**Comment by lusoe302** [comment]
Can I get unlocked file please upload . Thank You .
